File: Server/src/main/java/Server/core/game/content/consumable/effects/KegOfBeerEffect.java
package core.game.content.consumable.effects;

import core.game.node.entity.player.Player;
import core.game.content.consumable.ConsumableEffect;
import core.game.node.entity.skill.Skills;

public class KegOfBeerEffect extends ConsumableEffect {

    private final static int healing = 15;

    @Override
    public void activate(Player p) {
        final int attackLevelReduction = (p.getSkills().getLevel(Skills.ATTACK) * 65) / 99;
        final MultiEffect effect = new MultiEffect(new HealingEffect(15), new RandomSkillEffect(Skills.STRENGTH, 2, 10), new SkillEffect(Skills.ATTACK, attackLevelReduction, 0));
        effect.activate(p);
    }

    @Override
    public int getHealthEffectValue(Player player) {
        return healing;
    }
}

File: Server/src/main/kotlin/rs09/game/content/consumable/effects/KegOfBeerEffect.kt
package rs09.game.content.consumable.effects

import core.game.content.consumable.ConsumableEffect
import core.game.content.consumable.effects.HealingEffect
import core.game.content.consumable.effects.MultiEffect
import core.game.content.consumable.effects.SkillEffect
import core.game.node.entity.player.Player
import core.game.node.entity.skill.Skills


class KegOfBeerEffect : ConsumableEffect() {

    override fun activate(p: Player?) {
        val effect = MultiEffect(HealingEffect(15), SkillEffect(Skills.STRENGTH, 10.0, 0.0), SkillEffect(Skills.ATTACK, -40.0, 0.0))
        effect.activate(p)
    }

}